### PR TITLE
Java 11.3: Varargs and Ambiguity #2219

### LINKED ISSUE #2219: Java 11.3: Varargs and Ambiguity
### Description

Add audio, documentation, and video regarding the **Varargs and Ambiguity**.

____

### Location of File to be added

The files should be added inside the `Varargs and Ambiguity`  folder which is inside the `Varargs`  folder present in the   `Java`  main folder.  The documentation file can be in the `.md`/`.py`/`.ipynb` format. Audio should be in `.mp3` format and to be submitted through a Google drive link. Video can be in any format and should be submitted through a Google Drive link.

 **Note:**
 - You are required to comment with `\assign CONTENT_TYPE`, so that other content types can be created at the same time.
 - Failing to comment in the above manner will lead to the removal of assignment from the issue even if the bot assigns.
 - You are required to do a PR within 7 days otherwise you will be unassigned and others will be assigned.
 - Issues will be assigned on a first come first serve basis.

### Domain

Java_Domain

### Type of Contribution

Audio, Video, Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ishivamgupta9**
<hr>

## Description 📜

I have created documentation on  Java 11.3: Varargs and Ambiguity.
Fixes #2219  

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Java_Domain

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸
Screenshot (599)

<hr>

### CODE REVIEW

**Review by sumitaccess007 — CHANGES_REQUESTED** [review]
@ishivamgupta9 

The changes in this PR are fine now. Please check with below points -
1. Below line is not delivering any meaningful message, please try reading the sentence again and make corrections to this.
"Now if the interface variable uses the class implementing the interface it may confuse. This is the ambiguity problem in java."

2. Code Indentation is missing.
For the code snippet which you have provided as examples in Doc, make sure you indent them properly, for maximum code readability and quick understanding.

Feel free to ping in case of any confusions.

Thanks

**Comment by ishivamgupta9** [comment]
@sumitaccess007 

I have done the required changes could you please check it once.

**Review by sumitaccess007 — COMMENTED** [review]
@garvitraj 

Level-1 Review is fine for me.
I would request you to please have a final look for this PR.

Thanks

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Also, add a small paragraph about the places where varargs can be added like constructor and give individual examples for them

**Comment by AkMo3** [comment]
> 
> 
> Also, add a small paragraph about the places where varargs can be added like constructor and give individual examples for them

My bad, it is not required here!

File: Java/Varargs/Varargs_And_Ambiguity/Varargs_And_Ambiguity.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+```        
+ So here instead of overloading, we can store in an array as shown above and then traverse and add the result.
+
+  
+### Ambiguity
+
+A Java class can implement multiple interfaces and each interface can have some variables with respective name.
+Now, suppose a class implements two interfaces but both have one or more variables of same type and same variable name.
+Now when we call the variable of interface(i.e carModelName in the below code) then the compiler will give us an error because it will get confused between the similar variables of the interface. This is the ambiguity problem in java.
```
Comment (AkMo3) [inline Java/Varargs/Varargs_And_Ambiguity/Varargs_And_Ambiguity.md:88]: This is not the ambiguity. Ambiguity arises in cases like
```java
void foo(int num1, int... num2) {}

void foo(int.... num1) {}
```

Also in cases like:
```java
class SomeClass {
    void foo(int... num) {}

    void foo(boolean... isTrue) {} 
}

public static void main(String[] args) {
    foo(); // ambiguous
}

**Review by AkMo3 — CHANGES_REQUESTED** [review]

**Comment by ishivamgupta9** [comment]
@AkMo3 

I have done some changes. Please have a look and let me know if something more is needed.

**Pull request merged**